spec/python/run_benchmark_process_xor.py

```diff
@@ -4,20 +4,17 @@
 from timeit import default_timer as timer
 from benchmark_process_xor import BenchmarkProcessXor
 
-maxnum = 0
-
 t1 = timer()
 
 r = BenchmarkProcessXor.from_file("%s/benchmark_process_xor.dat" % os.environ['DATA_DIR'])
 
 t2 = timer()
 
 for chunk in r.chunks:
-    tm = max(chunk.body.numbers)
-    maxnum = max(tm, maxnum)
+    for n in chunk.body.numbers:
+        pass
```

Comment:
It skips doing any job => a clever enough compiler would just optimize away the whole loop => 0 time spent.

Reply:
CPython is not that clever, I assure you. :) 

Also entire code misses the point. If parsing is not lazy (and its not), there is no point measuring this. :)

Reply:
But some of the languages are. And Python compiler might become more intelligent in the future.

My rationale is very simple: "benchmark" solves a close-to-real-world task that is impossible to solve without doing certain time-consuming things, like going over whole parsed structure. If you'll have that summing/maxing in some language and would omit it in others, it's not a fair comparison.

Reply:
The way I understand it, the benchmark should measure parsing time. Right? That is the oneliner in `BenchmarkProcessXor.from_file(...)`. What this code measures is how Python is faster than C when processing dicts and lists.

Reply:
What you could have said instead is: Kaitai could become more lazy (when parsing) in the future. Only then iterating over the result object would make any kind of sense.

Reply:
That's valid argument too ;)